apis/rabbitmq/v1beta1/rabbitmquser_types.go

```diff
@@ -0,0 +1,134 @@
+	RabbitmqClusterName string `json:"rabbitmqClusterName"`
+
+	// +kubebuilder:validation:Optional
+	// VhostRef - reference to the RabbitMQVhost resource (defaults to default vhost "/" if empty)
+	VhostRef string `json:"vhostRef,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Username - the username in RabbitMQ (defaults to CR name)
+	Username string `json:"username,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Permissions - user permissions on the vhost
+	Permissions RabbitMQUserPermissions `json:"permissions,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Tags - RabbitMQ user tags
+	Tags []string `json:"tags,omitempty"`
+}
+
+// RabbitMQUserStatus defines the observed state of RabbitMQUser
+type RabbitMQUserStatus struct {
+	// Conditions
+	Conditions condition.Conditions `json:"conditions,omitempty" optional:"true"`
+
+	// ObservedGeneration - the most recent generation observed for this resource
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// SecretName - name of the secret containing user credentials
+	SecretName string `json:"secretName,omitempty"`
+
+	// Username - actual username used in RabbitMQ
+	Username string `json:"username,omitempty"`
+
+	// Vhost - actual vhost name used in RabbitMQ
+	Vhost string `json:"vhost,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:path=rabbitmqusers,shortName=rmquser,categories=all;rabbitmq
+//+kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".spec.rabbitmqClusterName"
+//+kubebuilder:printcolumn:name="Username",type="string",JSONPath=".status.username"
+//+kubebuilder:printcolumn:name="Vhost",type="string",JSONPath=".status.vhost"
+//+kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[0].status"
+//+kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[0].message"
+
+// RabbitMQUser is the Schema for the rabbitmqusers API
+type RabbitMQUser struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RabbitMQUserSpec   `json:"spec,omitempty"`
+	Status RabbitMQUserStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// RabbitMQUserList contains a list of RabbitMQUser
+type RabbitMQUserList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RabbitMQUser `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&RabbitMQUser{}, &RabbitMQUserList{})
+}
+
+// IsReady returns true if the user is ready
+func (instance RabbitMQUser) IsReady() bool {
+	return instance.Status.Conditions.IsTrue(condition.ReadyCondition)
+}
+
+const (
+	// UserFinalizer - finalizer to protect user from deletion when owned by TransportURL
+	UserFinalizer = "rabbitmquser.rabbitmq.openstack.org/finalizer"
+
+	// UserReadyCondition indicates that the user is ready
+	UserReadyCondition condition.Type = "UserReady"
```

Comment:
is this condition used in the service operators? If so I think we should probably prefix it with RabbitMQ. Otherwise it is not easy visible for the admin user to know what is the reference.